quantum/matrix.c

```diff
@@ -194,114 +194,61 @@ uint8_t matrix_key_count(void)
 
 static void init_cols(void)
 {
-    int B = 0, C = 0, D = 0, E = 0, F = 0;
-
 #if DIODE_DIRECTION == COL2ROW
-    for(int x = 0; x < MATRIX_COLS; x++) {
-        int col = COLS[x];
+    for (uint16_t col = 0; col < MATRIX_COLS; col++) {
+        const uint8_t col_pin = COL_PINS[col];
 #else
-    for(int x = 0; x < MATRIX_ROWS; x++) {
-        int col = ROWS[x];
+    for (uint16_t col = 0; col < MATRIX_ROWS; col++) {
+        const uint8_t col_pin = ROW_PINS[col];
```

Comment:
`col_pin` references would become `input_pin`, and `row_pin` references would become `output_pin` - the function names could change to reflect that as well. It'd also make it a little clearer how things are actually working.